### PR TITLE
divide up image utils

### PR DESCRIPTION
**Pull request by deads2k**
builds on https://github.com/openshift/origin/pull/22970 to divide imageutils into rough buckets of

 1. using internal APIs: pkg/image/internalimageutil
 2. oc only: pkg/oc/lib/ocimageutil
 3. probably shared: pkg/image/util

### CODE REVIEW

File: pkg/image/apiserver/registry/image/strategy.go [hunk 1]
```diff
@@ -99,7 +100,7 @@ func (s imageStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Ob
 	if newImage.DockerImageManifest != oldImage.DockerImageManifest {
 		ok := true
 		if len(newImage.DockerImageManifest) > 0 {
-			ok, err = util.ManifestMatchesImage(oldImage, []byte(newImage.DockerImageManifest))
+			ok, err = internalimageutil.ManifestMatchesImage(oldImage, []byte(newImage.DockerImageManifest))
```
Comment (soltysh) [inline pkg/image/apiserver/registry/image/strategy.go]: Dang, I'm gonna needs those somehow to work with external stuff to make the describers work :disappointed:

**Comment by soltysh** [comment]
/hold
@deads2k it looks like you have an older version of my PR, you'd need to rebase or wait when it lands in origin, it currently fails units

**Comment by deads2k** [comment]
rebased, retagged

**Pull request merged**